### PR TITLE
Implement bug reporting via email

### LINKED ISSUE #210: Implement user-friendly way to report bugs or request features via email
A normal app user has no GitHub account to report a bug or request a feature. Therefore a convenient way via email needs to be implemented. 

**Proposal:**
* Add new entry to about view
* Let the user choose between bug, feature and a general contact (merge with #209)
* Create email templates for each concern
* In case of a bug report, automatically extract Android and app version
* Implement possibility to send a bug report, if an import error occurs, including the JSON data which causes the issue (see #211)

### PR DESCRIPTION
**Pull request by chrgernoe**
Fixes #210

### CODE REVIEW

File: app/src/main/java/com/yacgroup/yacguide/AboutActivity.kt [hunk 1]
```diff
@@ -89,16 +80,18 @@ class AboutActivity : BaseNavigationActivity() {
...
-            action = Intent.ACTION_SEND
-            type = "plain/text"
-            putExtra(Intent.EXTRA_EMAIL, arrayOf(getString(R.string.contact_email_address)))
-        }
-        if (emailIndent.resolveActivity(packageManager) != null) {
-            startActivity(emailIndent)
-        } else {
-            Toast.makeText(this, R.string.no_email_app_available, Toast.LENGTH_SHORT).show()
+    private fun _selectContactConcern() {
+        val sendFeedback = ContactUtils(this)
```
Comment (paetz) [inline app/src/main/java/com/yacgroup/yacguide/AboutActivity.kt:84]: `sendFeedback` is not a proper variable name. Do you mean sth like `contactInterface`?

File: app/src/main/java/com/yacgroup/yacguide/utils/ActivityUtils.kt [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package com.yacgroup.yacguide.utils
+
+import android.content.pm.PackageManager
+import androidx.appcompat.app.AppCompatActivity
+
+open class ActivityUtils(var activity: AppCompatActivity) {
```
Comment (paetz) [inline app/src/main/java/com/yacgroup/yacguide/utils/ActivityUtils.kt:23]: - [ ] `activity` should be protected and constand (`val`?)

File: app/src/main/java/com/yacgroup/yacguide/utils/ContactUtils.kt [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+import java.io.FileOutputStream
+import java.text.SimpleDateFormat
+import java.util.*
+import kotlin.collections.ArrayList
+
+import com.yacgroup.yacguide.R
+
+class ContactUtils(activity: AppCompatActivity) : ActivityUtils(activity) {
+
+    private var _sdkVersion: Int = android.os.Build.VERSION.SDK_INT
```
Comment (paetz) [inline app/src/main/java/com/yacgroup/yacguide/utils/ContactUtils.kt:35]: - [ ] use `val`

File: app/src/main/java/com/yacgroup/yacguide/utils/ContactUtils.kt [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    private var _androidRelease: String = android.os.Build.VERSION.RELEASE
```
Comment (paetz) [inline app/src/main/java/com/yacgroup/yacguide/utils/ContactUtils.kt:36]: - [ ] use `val`

File: app/src/main/java/com/yacgroup/yacguide/utils/ContactUtils.kt [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+
+    fun reportImportError(errMsg: String, jsonFile: Uri) {
```
Comment (paetz) [inline app/src/main/java/com/yacgroup/yacguide/utils/ContactUtils.kt:38]: Why do you use two separate methods for reporting import error and bug? Most of the method bodies are equal.

File: app/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -150,6 +154,12 @@
         <item>Projektierte Felsen \u2117</item>
     </string-array>
 
+    <!-- IMPORTANT: This list must be in sync with the code in the AboutActivity class. -->
+    <string-array name="contactConcerns">
+        <item>Allgemeines Anliegen</item>
+        <item>Fehler melden</item>
```
Comment (paetz) [inline app/src/main/res/values/strings.xml:160]: - [ ] You have already defined this string above. Use it here

File: app/src/main/res/values/strings.xml [hunk 7]
```diff
@@ -53,13 +53,16 @@
     <string name="license">Lizenz</string>
     <string name="version">Version</string>
     <string name="contact">Kontakt</string>
+    <string name="contact_details">"Generelles Feedback und Fehler melden"</string>
```
Comment (paetz) [inline app/src/main/res/values/strings.xml:56]: - [ ] Remove quots

**Review by paetz — APPROVED** [review]

File: app/src/main/java/com/yacgroup/yacguide/utils/ContactUtils.kt (hunk 5 above)
Comment (chrgernoe) [reply app/src/main/java/com/yacgroup/yacguide/utils/ContactUtils.kt:38]: I can try to merge but I have the feeling that this results in a more complex code.

**Review by paetz — DISMISSED** [review]

**Comment by paetz** [comment]
Please trigger a new dev release after merging this.

**Pull request merged**